src/System.Private.CoreLib/src/System/Reflection/Emit/DynamicILGenerator.cs

```diff
@@ -1024,12 +1016,12 @@ internal int GetTokenFor(VarArgMethod varArgMethod)
             m_tokens.Add(varArgMethod);
             return m_tokens.Count - 1 | (int)MetadataTokenType.MemberRef;
         }
-        internal string GetString(int token) { return this[token] as string; }
+        internal string? GetString(int token) { return this[token] as string; }
 
-        internal byte[] ResolveSignature(int token, int fromMethod)
+        internal byte[]? ResolveSignature(int token, int fromMethod)
         {
             if (fromMethod == 0)
-                return (byte[])this[token];
+                return this[token] as byte[];
```

Comment:
Why did this change the type of cast used?  Couldn't it just be `return (byte[]?)this[token];`?

Reply:
Because `this[token]` is nullable, on row 1001:
```csharp
internal object? this[int token]
        {
            get
            {
                token &= 0x00FFFFFF;

                if (token < 0 || token > m_tokens.Count)
                    return null;

                return m_tokens[token];
            }
        }
```

Reply:
Right, but why change it to be `this[token] as byte[]` rather than just adding a question mark to make it `(byte[]?)this[token]`?

Reply:
Well if that is preferred then will update

Reply:
Thanks.  I prefer it for a few reasons.  First, and most importantly, it doesn't change the IL, whereas changing to use `as` does, and that means the risk of introducing a break goes way down.  Second, and less importantly, it's a smaller change.  Finally, just adding the `?` to the type name is an easy to recognize pattern when doing a code review and easy to confirm is non-breaking.

Reply:
Thank you updated this and all similar changes